pkg/trace/agent/agent.go

```diff
@@ -84,6 +86,13 @@ func NewAgent(ctx context.Context, conf *config.AgentConfig) *Agent {
 	sw := writer.NewStatsWriter(conf, statsChan)
 	svcW := writer.NewServiceWriter(conf, filteredServiceChan)
 
+	timerSet := timing.NewSet(ctx,
+		"datadog.trace_agent.internal.process_trace_ms",
+		"datadog.trace_agent.internal.concentrator_ms",
+		"datadog.trace_agent.internal.sample_ms",
```

Comment:
We also need one more timing metric, for `trace/writer/sender.doSend()`, `datadog.trace_agent.internal.sender.send_ms`, but this one needs to be tagged by the sender instance, because we have three of them: services, stats, traces

Reply:
Doesn't that already happened [here](https://github.com/DataDog/datadog-agent/blob/1117c66cb74011a398106f237c96dab447e5658c/pkg/trace/writer/trace.go#L119-L120) triggered from [here](https://github.com/DataDog/datadog-agent/blob/1117c66cb74011a398106f237c96dab447e5658c/pkg/trace/writer/payload.go#L136-L142). Or do you mean something else?

Reply:
Instead of a `_ms` suffix, can't we report them as second and instead set the proper unit as a metric metadata? (that's do-able in the UI for a given account, if we wanted to do it for all metrics that's a minor backend change we can do).

Reply:
I'm not sure I understand what you mean by using seconds. The reason I chose "ms" is because that's what our dogstatsd client uses too: https://godoc.org/github.com/DataDog/datadog-go/statsd#Client.Timing

Reply:
> Instead of a `_ms` suffix, can't we report them as second and instead set the proper unit as a metric metadata? (that's do-able in the UI for a given account, if we wanted to do it for all metrics that's a minor backend change we can do).

Yes, good point. That would be better.

Reply:
> Doesn't that already happen [here](https://github.com/DataDog/datadog-agent/blob/1117c66cb74011a398106f237c96dab447e5658c/pkg/trace/writer/trace.go#L119-L120) triggered from [here](https://github.com/DataDog/datadog-agent/blob/1117c66cb74011a398106f237c96dab447e5658c/pkg/trace/writer/payload.go#L136-L142)? Or do you mean something else?

Oh I didn't realize we had flush duration for all of them. Ok I think we're covered then. It would be a bit nicer to have the writer as a tag, but since we have all the metrics already then we don't need this.

Reply:
I guess one follow-up change later could be to standardize on using this metrics library for all internal metrics, and replace the current `sendStats` internal mini-pipeline with that.

Reply:
@LotharSee having looked deeper, if we mark these metrics as timings they will be treated as such and histograms will be created for them (95p, etc.). That's the very functionality we're somewhat trying to replicate here. So we'll have to stay as we are.